src/storage/index.ts

```diff
@@ -25,21 +27,143 @@ export const deleteConferenceSettings = () => {
 // Settings
 const SETTINGS_KEY = "SettingsKey";
 
+interface saveSettingFormat {
+  active: boolean;
+  encoded: string;
+  id: number;
+}
+
 export const saveSettings = (settings: AppSettings) => {
   const encoded = Buffer.from(JSON.stringify(settings), "utf-8").toString(
     "base64"
   );
 
-  localStorage.setItem(SETTINGS_KEY, encoded);
+  const str = localStorage.getItem(SETTINGS_KEY);
+  if (str) {
+    const parsed = JSON.parse(str);
+
+    if (parsed.length < 1) {
+      localStorage.setItem(
+        SETTINGS_KEY,
+        JSON.stringify([{ id: 1, encoded, active: true }])
+      );
+    } else {
+      localStorage.setItem(
+        SETTINGS_KEY,
+        JSON.stringify([
+          ...parsed,
+          {
+            encoded,
+            active: false,
+            id: parsed.length + 1,
+          },
+        ])
+      );
+    }
+  } else {
+    localStorage.setItem(
+      SETTINGS_KEY,
+      JSON.stringify([{ id: 1, encoded, active: true }])
+    );
+  }
 };
 
-export const getSettings = (): AppSettings => {
+export const editSettings = (settings: AppSettings, id: number) => {
+  const encoded = Buffer.from(JSON.stringify(settings), "utf-8").toString(
+    "base64"
+  );
+
+  const str = localStorage.getItem(SETTINGS_KEY);
+  if (str) {
+    const parsed = JSON.parse(str);
+
+    // for edit:
+    const newData = parsed.map((el: saveSettingFormat) => {
+      if (el.id === id)
+        return {
+          id: el.id,
+          active: el.active,
+          encoded: encoded,
+        };
+      else return el;
+    });
+
+    localStorage.setItem(SETTINGS_KEY, JSON.stringify(newData));
+  }
+};
+export const setActiveSettings = (id: number) => {
   const str = localStorage.getItem(SETTINGS_KEY);
   if (str) {
-    const planText = Buffer.from(str, "base64").toString("utf-8");
-    return JSON.parse(planText) as AppSettings;
+    const parsed = JSON.parse(str);
```

Comment:
const newData = parsed.map((el: saveSettingFormat) => ({
  ...el,
  active: el.id === id,
}));

localStorage.setItem(SETTINGS_KEY, JSON.stringify(newData));

maybe this much simpler, could you rethink about this?

Reply:
ok, will do